windows/RNFS/RNFSManager.cpp

```diff
@@ -0,0 +1,1115 @@
+            {
+                content.Headers().TryAppendWithoutValidation(winrt::to_hstring(header.first), winrt::to_hstring(header.second.AsString()));
+            }
+        }
+        request.Content(content);
+
+        co_await m_tasks.Add(jobId, ProcessDownloadRequestAsync(promise, request, filePath, jobId, progressInterval, progressDivider));
+    }
+    catch (const hresult_error& ex)
+    {
+        // "Failed to download file." 
+        promise.Reject(winrt::to_string(ex.message()).c_str());
+    }
+    m_tasks.Cancel(jobId);
+}
+
+
+winrt::fire_and_forget RNFSManager::uploadFiles(RN::JSValueObject options, RN::ReactPromise<RN::JSValueObject> promise) noexcept
+{
+    auto jobId{ options["jobId"].AsInt32() };
+    try
+    {
+        auto method{ options["method"].AsString() };
+
+        winrt::Windows::Web::Http::HttpMethod httpMethod{ winrt::Windows::Web::Http::HttpMethod::Post() };
+        if (method.compare("POST") != 0)
+        {
+            if (method.compare("PUT") == 0)
+            {
+                httpMethod = winrt::Windows::Web::Http::HttpMethod::Put();
+            }
+            else
+            {
+                promise.Reject("Invalid HTTP request: neither a POST nor a PUT request.");
+                co_return;
+            }
+        }
+
+        auto const& files{ options["files"].AsArray() };
+        uint64_t totalUploadSize = 0;
+        for (const auto& fileInfo : files)
+        {
+            auto const& fileObj{ fileInfo.AsObject() };
+            auto filepath{ fileObj["filepath"].AsString() };
+
+            winrt::hstring directoryPath, fileName;
+            splitPath(filepath, directoryPath, fileName);
+
+            try
+            {
+                StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(directoryPath) };
+                StorageFile file{ co_await folder.GetFileAsync(fileName) };
+                auto fileProperties{ co_await file.GetBasicPropertiesAsync() };
+                totalUploadSize += fileProperties.Size();
+            }
+            catch (...)
+            {
+                continue;
+            }
+        }
+        if (totalUploadSize <= 0)
+        {
+            promise.Reject("No files to upload");
+            co_return;
+        }
+
+        co_await m_tasks.Add(jobId, ProcessUploadRequestAsync(promise, options, httpMethod, files, jobId, totalUploadSize));
+    }
+    catch (const hresult_error& ex)
+    {
+        // "Failed to upload file."
+        promise.Reject(winrt::to_string(ex.message()).c_str());
+    }
+    m_tasks.Cancel(jobId);
+}
+
+
+void RNFSManager::touch(std::string filepath, int64_t mtime, int64_t ctime, bool modifyCreationTime, RN::ReactPromise<std::string> promise) noexcept
+try
+{
+
+    std::filesystem::path path(filepath);
+    path.make_preferred();
+    auto s_path{ path.c_str() };
+    LPCWSTR actual_path{ s_path };
```

Comment:
is this needed? if so drop the L, just PCWSTR is fine (L is a now deprecated / no-op type prefix from 16 bit Windows :)  )

Reply:
Tested this method using just PCWSTR, and I'm still getting the same results. I made that change in a recent commit.

On the topic of whether or not it is needed...
Android doesn't allow for users to update the file creation time, but iOS does. I thought it would be better to try and replicate the same functionality of iOS for this method.

I couldn't find a way to update the file creation time in WinRT, so I decided to use CreateFile2() from Win32. 

This Win32 method should still work across all Windows 10 devices. Should I remove the ability to update the creation time of a file so that only WinRT is used?